identity/src/main/kotlin/net/thechance/identity/security/IpRateLimitingFilter.kt

```diff
@@ -0,0 +1,39 @@
+package net.thechance.identity.security
+
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import net.thechance.identity.security.config.RateLimitProperties
+import net.thechance.identity.service.RateLimitManagerService
+import org.springframework.http.HttpStatus
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+
+@Component
+class IpRateLimitingFilter(
+    private val rateLimitProperties: RateLimitProperties,
+    private val rateLimitManagerService: RateLimitManagerService
+) : OncePerRequestFilter() {
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        val authHeader = request.getHeader("Authorization")
+        val requestUri = request.requestURI
+        val clientIp = request.remoteAddr
```

Comment:
this is not real clientIp please check this thread: https://thechance-workspace.slack.com/archives/C091WT8R8LR/p1759257452895629